tests/v4l2decode.cpp

```diff
@@ -330,6 +329,153 @@ static bool displayOneVideoFrameAndroid(int32_t fd, int32_t index)
 
     return true;
 }
+
+#elif __ENABLE_WAYLAND__
+struct display {
+    SharedPtr<wl_display>        display;
```

Comment:
not so good, you can't decide the order to destroy these object.
if wl_display is destroyed before wl_surface, there will be exception.